phone.py

```diff
@@ -1,33 +1,33 @@
 import wave
 from time import sleep
-
+import PIL.Image     # run pip install Pillow
+import PIL.ImageTk
 import cv2
 import pyaudio
 
 from playsound import playsound
 
-
 from AudioStreamer import AudioStreamerListener
 from EasySocket import EasySocketClient
 
 ip_client = "127.0.0.1"
 chunk = 1024
 
-
 win_name = "DOORBELL"
 
-def on_video_received(data):
-    cv2.imshow(win_name, data)
+video_frame = None
```

Comment:
ici, si tu veux éviter des problèmes d'écriture/lecture concurrentiel, il serait interessant de protéger cette variable avec un semaphore
le top du top, c'est de construire une classe VideoFrameThreadSafe (par exemple) dans laquelle tu stockes la frame avec 2 methodes : 
get_frame() et set_frame(frame)
tu utilises l'objet :  threading.Lock()

voici un petit exemple que j'ai trouvé : 
`mutex = threading.Lock()  # is equal to threading.Semaphore(1)

def fun1():
    while True:
        with mutex:
            print(1)
        time.sleep(.5)`

Reply:
pour comprendre les threads en python  : https://www.geeksforgeeks.org/multithreading-python-set-1/
https://www.geeksforgeeks.org/multithreading-python-set-2/